backend/main.go

```diff
@@ -146,44 +159,71 @@ func (s *Server) handleGetTeam(w http.ResponseWriter, r *http.Request) {
 		http.NotFound(w, r)
 		return
 	}
-	enc := json.NewEncoder(w)
-	w.Header().Set("Content-Type", "application/json")
-	enc.Encode(team)
+
+	if s.auth.CanActOnTeam(r.Context().Value("user").(auth.User), team, auth.ActionRead) {
+		enc := json.NewEncoder(w)
+		w.Header().Set("Content-Type", "application/json")
+		enc.Encode(team)
+	} else {
+		http.Error(w, "You are not authorized to view this team.", http.StatusForbidden)
+	}
 }
 
 func (s *Server) handlePostTeam(w http.ResponseWriter, r *http.Request) {
 	team, ok := readTeamFromBody(w, r)
 	if !ok {
 		return
 	}
-	if !s.ensureTeamExistence(w, r, team.ID, false) {
+	_, isNew := s.ensureTeamExistence(w, r, team.ID, false)
+	if !isNew {
 		return
 	}
 	ctx, cancel := context.WithTimeout(r.Context(), s.storeTimeout)
 	defer cancel()
-	err := s.store.CreateTeam(ctx, team)
+	settings, err := s.store.GetSettings(ctx)
 	if err != nil {
-		log.Printf("Could not create team: error: %s", err)
-		http.Error(w, "Could not create team (see server log)", http.StatusInternalServerError)
+		http.Error(w, "Authorization failed due to internal server error", http.StatusInternalServerError)
 		return
 	}
+	permissions := settings.GeneralPermissions
+	if s.auth.CanActOnTeamList(r.Context().Value("user").(auth.User), permissions, auth.ActionAddTeam) {
+		if reflect.DeepEqual(team.Permissions, models.TeamPermissions{}) {
+			team.Permissions.Read = permissions.ReadTeamList
+			team.Permissions.Write = permissions.AddTeam
+		}
+
+		err := s.store.CreateTeam(ctx, team)
+		if err != nil {
+			log.Printf("Could not create team: error: %s", err)
+			http.Error(w, "Could not create team (see server log)", http.StatusInternalServerError)
```

Comment:
I have left it for now because this was part of Andrew's implementation but it would probably make sense to implement a more useful log application wide in the future.

Reply:
Yes, I filed #66 for this. This is fine for now.